### PR TITLE
Enable generation and deploy from container

### PR DESCRIPTION
**Pull request by fbertsch**
@acmiyaguchi this roughly matches the way we deploy schemas for MSG. You will probably recognize some of the code in `bin/generate`. I've tested this a few times, on a previously existing publish branch [here](https://github.com/mozilla/looker-hub/tree/test-lookml-generation), and on a new publish branch [here](https://github.com/mozilla/looker-hub/tree/test-generation-2). The base branch `base` is, right now, just a copy of `main`.

We'll need to get SRE involved for the dockerhub deploy, they may have us switch to GCR.

### CODE REVIEW

**Comment by fbertsch** [comment]
Resolves https://github.com/mozilla/lookml-generator/issues/5.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+version: '2'
```
Comment (acmiyaguchi) [inline docker-compose.yml:2]: Latest version is 3

File: bin/generate [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+#   GCLOUD_SERVICE_KEY: The service key used to authenticate with GCP.
+#                       Needs to have read access to all tables specified
+#                       in `namespaces.yaml`. Not required in local dev.
+#
+# Example usage:
+#   export HUB_SSH_KEY_BASE64=$(cat ~/.ssh/id_rsa | base64)
+#   make build && make run
+
+HUB_REPO_URL=${HUB_REPO_URL:-"git@github.com:mozilla/looker-hub.git"}
+HUB_BRANCH_SOURCE=${HUB_BRANCH_SOURCE:-"base"}
```
Comment (acmiyaguchi) [inline bin/generate:30]: `main`?

File: bin/generate [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+  popd
+}
+
+function main() {
+  pushd .
+  cd /app
+
+  # don't print auth setup
+  set -x
```
Comment (acmiyaguchi) [inline bin/generate:136]: set +x?

**Review by acmiyaguchi — COMMENTED** [review]
This seems reasonable to me. Have you run this to generate a testing branch? I don't see a testing branch in this repo at the moment.

File: bin/generate (hunk 2 above)
Comment (fbertsch) [reply bin/generate:30]: Looker is configured to deploy from `main`, so when we publish this the first time, the `main` branch will have the generated schemas + files from base, and base will have the things we haven't yet gotten around to adding to the generation pipeline.

**Review by acmiyaguchi — APPROVED** [review]
Looks good for scheduling.

**Pull request merged**